catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/MatchAnyValidator.java

```diff
@@ -37,42 +39,62 @@ public MatchAnyValidator(List<AttributeValidator> validators) {
       return Optional.empty();
     }
 
-    List<Optional<AttributeValidationReport>> validationReportList =
-        validators
-            .stream()
-            .map(validator -> validator.validate(attribute))
-            .collect(Collectors.toList());
+    List<Serializable> serializables = attribute.getValues();
+    List<Optional<AttributeValidationReport>> resultValidationReportList = new ArrayList<>();
 
-    return generateValidationReports(validationReportList);
+    for (Serializable serializable : serializables) {
+
+      List<Optional<AttributeValidationReport>> validationReportList =
+          validateAttributeSerializable(attribute.getName(), serializable);
+
+      if (validationReportList.size() == validators.size()) {
+        resultValidationReportList.addAll(validationReportList);
+      }
+    }
+
+    return generateValidationReports(resultValidationReportList);
+  }
+
+  private List<Optional<AttributeValidationReport>> validateAttributeSerializable(
+      String attributeName, Serializable serializable) {
+    List<Optional<AttributeValidationReport>> validationReportList = new ArrayList<>();
+    for (AttributeValidator attributeValidator : validators) {
+      Attribute newAttribute = new AttributeImpl(attributeName, serializable);
+      Optional<AttributeValidationReport> attributeValidationReport =
+          attributeValidator.validate(newAttribute);
+      if (attributeValidationReport.isPresent()) {
+        validationReportList.add(attributeValidationReport);
+      }
+    }
+    return validationReportList;
   }
 
   private Optional<AttributeValidationReport> generateValidationReports(
       List<Optional<AttributeValidationReport>> validationReportList) {
 
+    if (CollectionUtils.isEmpty(validationReportList)) {
+      return Optional.empty();
+    }
+
     AttributeValidationReportImpl result = new AttributeValidationReportImpl();
 
     for (Optional<AttributeValidationReport> attributeValidationReportOptional :
         validationReportList) {
 
-      if (attributeValidationReportOptional.isPresent()) {
```

Comment:
Why was this check removed? `Optional.get` is called on the next line; where its the guarantee that `attributeValidationReportOptional` is present?

Reply:
https://github.com/codice/ddf/pull/2902/files#diff-405452c32478dac410dedc3ab58fd0e9R65
It's only added to the list if present

Reply:
Then why is `validateAttributeSerializable` even returning a `List<Optional<AttributeValidationReport>>` instead of a `List<AttributeValidationReport>`?

Reply:
good point; fixed